src/ansi-c/c_typecheck_base.cpp

```diff
@@ -148,11 +148,8 @@ void c_typecheck_baset::typecheck_new_symbol(symbolt &symbol)
     {
       // we don't need the identifiers
       code_typet &code_type=to_code_type(symbol.type);
-      for(code_typet::parameterst::iterator
-          it=code_type.parameters().begin();
-          it!=code_type.parameters().end();
-          it++)
-        it->set_identifier(irep_idt());
+      for(auto &parameter : code_type.parameters())
+        parameter.set_identifier(irep_idt());
```

Comment:
In a second commit in this PR maybe get rid of the `code_type` named temporary? And the block around these lines?

Reply:
done